core/state_transition.go

```diff
@@ -439,15 +445,55 @@ func (st *StateTransition) TransitionDb() (*ExecutionResult, error) {
 	// - reset transient storage(eip 1153)
 	st.state.Prepare(rules, msg.From, st.evm.Context.Coinbase, msg.To, vm.ActivePrecompiles(rules), msg.AccessList)
 
+	if !contractCreation {
+		// Increment the nonce for the next transaction
+		st.state.SetNonce(msg.From, st.state.GetNonce(sender.Address())+1)
+	}
+
+	// Check authorizations list validity.
+	if msg.AuthList != nil {
+		for _, auth := range msg.AuthList {
+			// Verify chain ID is 0 or equal to current chain ID.
+			if auth.ChainID != 0 && st.evm.ChainConfig().ChainID.Uint64() != auth.ChainID {
+				continue
+			}
+			authority, err := auth.Authority()
+			if err != nil {
+				continue
+			}
+			// Check the authority account 1) doesn't have code or has exisiting
+			// delegation 2) matches the auth's nonce
+			st.state.AddAddressToAccessList(authority)
```

Comment:
Might be nothing, but shouldn't we add the address to the access list after the nonce overflow check? afaict, you moved the nonce overflow check to L471 while it is the second step in the eip, so if the nonce is set to 2**64 -1 we will add the authority to the access list, while other clients might not. cc @marioevz we should create a test for that

Reply:
We don't have an explicit test for this, but we do call the authorization signer account during the transaction where the authorization failed due to the overflow nonce check here:
https://github.com/ethereum/execution-spec-tests/blob/f5799c8b89235718239f05e552e52e8c597763fb/tests/prague/eip7702_set_code_tx/test_set_code_txs.py#L2408-L2413
However if this did not catch the issue, would we need to add something like an `EXTCODE*` instead of calling?

Reply:
I think there is a test for this and the move to L471 fixed it :)